src/commonMain/kotlin/fr/acinq/lightning/wire/LightningMessages.kt

```diff
@@ -154,11 +155,22 @@ data class EncryptedChannelData(val data: ByteVector) {
     }
 }
 
-interface HasEncryptedChannelData : LightningMessage {
+interface HasEncryptedChannelData : HasChannelId {
     val channelData: EncryptedChannelData
-    fun withChannelData(data: ByteVector): HasEncryptedChannelData = withChannelData(EncryptedChannelData(data))
-    fun withChannelData(ecd: EncryptedChannelData): HasEncryptedChannelData = if (ecd.isEmpty()) this else withNonEmptyChannelData(ecd)
     fun withNonEmptyChannelData(ecd: EncryptedChannelData): HasEncryptedChannelData
+    fun withChannelData(data: ByteVector, logger: Logger? = null): HasEncryptedChannelData = withChannelData(EncryptedChannelData(data), logger)
```

Comment:
This is a bit convoluted but I don't have better to propose. Ideally `HasEncryptedChannelData` should be a `sealed class` and `withNonEmptyChannelData` a `protected abstract fun`, but double inheritance is forbidden so we are stuck with an interface.